tt_metal/llrt/wormhole/wh_hal_tensix_mem_map.cpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#if defined(ARCH_WORMHOLE_B0)
```

Comment:
I would like to eventually remove ARCH_NAME from the build system, any problem compiling this code for all architectures?

Reply:
Are you only doing this to cut down on compilation time?

Reply:
> I would like to eventually remove ARCH_NAME from the build system, any problem compiling this code for all architectures?

Yes, the hal is a step towards that goal, but unfortunately we end up w/ conflicts presently because the hal is incomplete.  The ifdefs are necessary to prevent duplicate symbols since, for example, some places still include GS specific headers so if compiling for WH we clash

Reply:
ok great